pallets/collator-assignment/src/lib.rs

```diff
@@ -159,16 +167,117 @@ pub mod pallet {
     }
 
     impl<T: Config> Pallet<T> {
+        /// Takes the bulk paras (parachains) and pool paras (parathreads)
+        /// and checks if the quantity of each is according to the
+        /// number of cores available.
+        /// If not, then both are sorted by the tip.
+        /// After that it wires through `order_paras` which is
+        /// the default behaviour when we are running in parachain
+        /// mode.
+        pub(crate) fn order_paras_with_core_config(
+            mut bulk_paras: Vec<ChainNumCollators>,
+            mut pool_paras: Vec<ChainNumCollators>,
+            core_allocation_configuration: &CoreAllocationConfiguration,
+            target_session_index: T::SessionIndex,
+            number_of_collators: u32,
+            collators_per_container: u32,
+            collators_per_parathread: u32,
+        ) -> (Vec<ChainNumCollators>, bool) {
+            let core_count = core_allocation_configuration.core_count;
+            let max_number_of_bulk_paras = core_allocation_configuration
+                .max_parachain_percentage
+                .mul(core_count);
```

Comment:
Why not saturating_mul?

```suggestion
                .saturating_mul(core_count);
```

Reply:
Reason being this achieves same thing as `saturating_mul` but without overhead of converting to and from `PerBill`.

See this source code: 
```rust
		/// Non-overflow multiplication.
		///
		/// This is tailored to be used with a balance type.
		impl<N> ops::Mul<N> for $name
		where
			N: Clone + UniqueSaturatedInto<$type> + ops::Rem<N, Output=N>
				+ ops::Div<N, Output=N> + ops::Mul<N, Output=N> + ops::Add<N, Output=N> + Unsigned,
			$type: Into<N>,
		{
			type Output = N;
			fn mul(self, b: N) -> Self::Output {
				overflow_prune_mul::<N, Self>(b, self.deconstruct(), Rounding::NearestPrefDown)
			}
		}
```

Reply:
Ah, I see. So `max_parachain_percentage` is `PerBill`, which means between 0 and 1, and multiplying by an `usize` can never overflow.

*disregard*